env.go

```diff
@@ -0,0 +1,177 @@
+			if value, ok := osenv[name]; ok {
+				env[name] = value
+			}
+
+			// EnvVars overrides provided by customer, highest priority
+			if len(overrides) > 0 {
+				if value, ok := overrides[name]; ok {
+					env[name] = value
+				}
+			}
+		}
+	}
+
+	return env
+
+}
+
+func getEnvDefaults(function *cloudformation.AWSServerlessFunction) map[string]string {
+
+	creds := getSessionOrDefaultCreds()
+
+	// Variables available in Lambda execution environment for all functions (AWS_* variables)
+	return map[string]string{
+		"AWS_SAM_LOCAL":                   "true",
+		"AWS_REGION":                      creds["region"],
+		"AWS_DEFAULT_REGION":              creds["region"],
+		"AWS_ACCESS_KEY_ID":               creds["key"],
+		"AWS_SECRET_ACCESS_KEY":           creds["secret"],
+		"AWS_SESSION_TOKEN":               creds["sessiontoken"],
+		"AWS_LAMBDA_FUNCTION_MEMORY_SIZE": strconv.Itoa(int(function.MemorySize)),
+		"AWS_LAMBDA_FUNCTION_TIMEOUT":     strconv.Itoa(int(function.Timeout)),
+		"AWS_LAMBDA_FUNCTION_HANDLER":     function.Handler,
+		// "AWS_ACCOUNT_ID=",
+		// "AWS_LAMBDA_EVENT_BODY=",
+		// "AWS_REGION=",
+		// "AWS_LAMBDA_FUNCTION_NAME=",
+		// "AWS_LAMBDA_FUNCTION_VERSION=",
+	}
+
+}
+
+func getEnvOverrides(logicalID string, filename string) map[string]string {
+
+	if len(filename) > 0 {
+
+		data, err := ioutil.ReadFile(filename)
+		if err != nil {
+			log.Printf("Could not read environment overrides from %s: %s\n", filename, err)
+			return map[string]string{}
+		}
+
+		// This is a JSON of structure {FunctionName: {key:value}, FunctionName: {key:value}}
+		overrides := map[string]map[string]string{}
+		if err = json.Unmarshal(data, &overrides); err != nil {
+			log.Printf("Invalid environment override file %s: %s\n", filename, err)
+			return map[string]string{}
+		}
+
+		return overrides[logicalID]
+
+	}
+
+	return map[string]string{}
+
+}
+
+func getSessionOrDefaultCreds() map[string]string {
+
+	region := "us-east-1"
+	key := "defaultkey"
+	secret := "defaultsecret"
+	sessiontoken := "sessiontoken"
+
+	result := map[string]string{
+		"region":  region,
+		"key":     key,
+		"secret":  secret,
+		"session": sessiontoken,
+	}
+
+	// Obtain AWS credentials and pass them through to the container runtime via env variables
+	if sess, err := session.NewSession(); err == nil {
+		if creds, err := sess.Config.Credentials.Get(); err == nil {
+			if *sess.Config.Region != "" {
+				result["region"] = *sess.Config.Region
+			}
+
+			result["key"] = creds.AccessKeyID
+			result["secret"] = creds.SecretAccessKey
+			result["sessiontoken"] = creds.SessionToken
+		}
+	}
+	return result
+}
+
+func getEnvFromOS() map[string]string {
+
+	result := map[string]string{}
+	for _, value := range os.Environ() {
+		keyVal := strings.Split(value, "=")
+		result[keyVal[0]] = keyVal[1]
+	}
+
+	return result
+}
+
+// Converts the input to string if it is a primitive type, Otherwise returns nil
+func toStringMaybe(value interface{}) (string, bool) {
```

Comment:
I'm not a fan of generic util packages, as they tend to become a random assortment of junk over time in most projects.

If the env stuff was more substantial it would probably warrant its own package, but as it is, I'd rather leave it.